framework/src/functions/LinearCombinationFunction.C

```diff
@@ -61,3 +61,21 @@ LinearCombinationFunction::value(Real t, const Point & p)
     val += _w[i] * _f[i]->value(t, p);
   return val;
 }
+
+RealGradient
+LinearCombinationFunction::gradient(Real t, const Point & p)
+{
+  RealGradient g;
+  for (unsigned i = 0; i < _f.size(); ++i)
+    g += _w[i] * _f[i]->gradient(t, p);
+  return g;
+}
+
+RealVectorValue
+LinearCombinationFunction::vectorValue(Real t, const Point & p)
```

Comment:
@aeslaughter - It looks like you added the `vectorValue()` method 4 years ago. Can you recall why this was needed and the `gradient()` method didn't suffice? I just want to make sure we don't have users duplicating implementations for these two methods.